src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerServiceTest.java

```diff
@@ -70,4 +82,106 @@ void sendMessage_actionOnResource_messageIsSentWithCorrectArguments(KafkaAction
     assertThat(messageArgument.getValue().getResource()).isEqualTo(timeEntry);
     assertThat(messageArgument.getValue().getAction()).isEqualTo(action);
   }
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE", "DELETE"})
+  void CompletableFutureReporting_actionOnResource_onFailureMessageLogged(KafkaAction action) throws ExecutionException, InterruptedException {
+    ReflectionTestUtils.setField(kafkaProducerService, "topicName", TOPIC_NAME);
+    ReflectionTestUtils.setField(kafkaProducerService, "projectVersion", "1.0.0");
+
+    UUID ownerId = UUID.fromString("ec703cac-de76-49c8-b1c4-83da6f8b42ce");
+    LocalDateTime actualStartTime = LocalDateTime.of(
+        2022, 1, 1, 9, 0, 0);
+    TimeEntry timeEntry = createTimeEntry(ownerId, getAsDate(actualStartTime));
+    String messageKey = generateMessageKey(timeEntry);
```

Comment:
nit: I think you could also move this block to a beforeEach or something, as its used across all these tests.

Reply:
Actioned